lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+short_description: Set Quotas on XFS filesystems
+description:
+  - Configure Quotas on XFS filesystems. /etc/projects and /etc/projid needs to be configured before calling this module when setting project quotas.
+version_added: "2.8"
+author: "William Leemans (@bushvin) <willie@elaba.net>"
+options:
+  type:
+    description:
+      - The xfs quota type.
+    type: str
+    required: true
+    choices:
+      - user
+      - group
+      - project
+  name:
+    description:
+      - The name of the user, group or project to apply the quota to, if other than default.
+    type: str
+  mountpoint:
+    description:
+      - the mountpoint on which to apply the quotas
+    type: str
+    required: true
+  bhard:
+    description:
+      - Hard blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  bsoft:
+    description:
+      - Soft blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  ihard:
+    description:
+      - Hard inodes quota limit.
+    type: int
+  isoft:
+    description:
+      - Soft inodes quota limit.
+    type: int
+  rtbhard:
+    description:
+      - Hard realtime blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  rtbsoft:
+    description:
+      - Soft realtime blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  state:
+    description:
+      - Whether to apply the limits or remove them.
+      - When removing limit, they are set to 0, and not quite removed.
+    type: str
+    default: present
+    choices:
+      - present
+      - absent
+
+requirements:
+   - xfsprogs
+'''
+
+EXAMPLES = r'''
+- name: Set default project soft and hard limit on /opt of 1g
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import grp
+import json
+import os
+import pwd
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            bhard=dict(type='str'),
+            bsoft=dict(type='str'),
+            ihard=dict(type='int'),
+            isoft=dict(type='int'),
+            mountpoint=dict(type='str', required=True),
+            name=dict(type='str'),
+            rtbhard=dict(type='str'),
+            rtbsoft=dict(type='str'),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            type=dict(type='str', required=True, choices=['user', 'group', 'project'])
+        ),
+        supports_check_mode=True
```

Comment:
We tend to keep trailing commas in multi-line lists and dicts for the last element.
```suggestion
        supports_check_mode=True,
```